src/test/java/org/jetbrains/kotlinx/lincheck/test/verifier/linearizability/MSQueueTest.kt

```diff
@@ -0,0 +1,66 @@
+package org.jetbrains.kotlinx.lincheck.test.verifier.linearizability
+
+import org.jetbrains.kotlinx.lincheck.Options
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import org.jetbrains.kotlinx.lincheck.test.AbstractLincheckTest
+import org.jetbrains.kotlinx.lincheck.verifier.linearizability.LinearizabilityVerifier
+import java.util.*
+import java.util.concurrent.atomic.AtomicReference
+
+class MSQueueTest : AbstractLincheckTest() {
+    private val q = MSQueue<Int>()
+
+    @Operation
+    fun push(value: Int) = q.push(value)
+
+    @Operation
+    fun pop() = q.pop()
+
+    override fun <O : Options<O, *>> O.customize() {
+        verifier(LinearizabilityVerifier::class.java)
+        sequentialSpecification(SequentialQueue::class.java)
+        requireStateEquivalenceImplCheck(false)
+    }
+}
+
+class SequentialQueue {
+    private val q = ArrayDeque<Int>()
+
+    fun push(value: Int) {
+        q.offer(value)
+    }
+
+    fun pop() = q.poll()
+    fun pop(ignore: Int) = pop()
```

Comment:
why do you need this method?

Reply:
I use it in DurableMSQueueTest. I need `ignore` param as signatures of sequential and parallel implementations must be the same.